libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java

```diff
@@ -1124,11 +993,87 @@ public void onClick(DialogInterface dialog, int id) {
     public void onActivityResult(int requestCode, int resultCode, Intent data) {
         super.onActivityResult(requestCode, resultCode, data);
 
-        if ((requestCode == LinkDialogFragment.LINK_DIALOG_REQUEST_CODE_ADD ||
-                requestCode == LinkDialogFragment.LINK_DIALOG_REQUEST_CODE_UPDATE)) {
-            // TODO: handle link/unlink
-        } else if (requestCode == ImageSettingsDialogFragment.IMAGE_SETTINGS_DIALOG_REQUEST_CODE) {
-            // TODO: handle media settings
+        if (requestCode == ImageSettingsDialogFragment.IMAGE_SETTINGS_DIALOG_REQUEST_CODE) {
+            if (data == null || data.getExtras() == null && tappedImagePredicate != null) {
+                return;
+            }
+
+            Bundle extras = data.getExtras();
+            JSONObject meta;
+            try {
+                meta = new JSONObject(StringUtils.notNullStr(extras.getString("imageMeta")));
+            } catch (JSONException e) {
+                // ignore errors
+                return;
+            }
+
+            AztecAttributes attributes = content.getElementAttributes(tappedImagePredicate);
```

Comment:
It seems that `tappedImagePredicate` is introduced only to have a means to retrieve the element in `onActivityResult` so, it needs to be saved in `onSaveInstanceState` as well. Alternatively, and since `onActivityResult` already makes heavy use of data passed via metadata (`meta`), I would suggest passing the required ID via that instead of a Fragment instance variable. Besides, the less instance variables we have, the better. Feel free to open a separate ticket to tackle that if not in this PR. Thanks!